ktor-client/ktor-client-features/ktor-client-auth/common/src/io/ktor/client/features/auth/Auth.kt

```diff
@@ -41,11 +41,24 @@ public class Auth(
                 if (origin.request.attributes.contains(circuitBreaker)) return@intercept origin
 
                 var call = origin
-                val candidateProviders = HashSet(feature.providers).apply { removeAll(feature.alwaysSend) }
+                val candidateProviders = HashSet(feature.providers).apply {
+                    removeAll(feature.alwaysSend - feature.providers.find { it is BearerAuthProvider })
```

Comment:
Can we avoid having a feature that knows about specific implementations of providers? It is a good sign that the feature is not flexible enough to support user's custom implementations. I would prefer to add extension points to the feature itself.

Reply:
Yep, will fix